commands/droplets.go

```diff
@@ -462,33 +463,60 @@ func RunDropletDelete(c *CmdConfig) error {
 		return err
 	}
 
-	if len(c.Args) < 1 && tagName == "" {
-		return doctl.NewMissingArgsErr(c.NS)
-	} else if len(c.Args) > 0 && tagName != "" {
-		return fmt.Errorf("please specify droplets identifiers or a tag name")
-	} else if tagName != "" {
-		if force || AskForConfirm("delete droplet by \""+tagName+"\" tag") == nil {
-			return ds.DeleteByTag(tagName)
-		}
-		return nil
-	}
+	deleteFn := func(ids []int) error {
+		if !force {
 
-	if force || AskForConfirm("delete droplet(s)") == nil {
+			// If we aren't forcing, print out the list
+			fmt.Println("Following Droplet(s) will be deleted:")
+
+			var list do.Droplets
+			for _, id := range ids {
+				d, err := ds.Get(id)
+				if err != nil {
+					return err
+				}
+				list = append(list, *d)
+			}
+
+			item := &droplet{droplets: list}
+			c.Display(item)
+		}
 
-		fn := func(ids []int) error {
+		if force || AskForConfirm("delete droplet(s)") == nil {
 			for _, id := range ids {
 				if err := ds.Delete(id); err != nil {
 					return fmt.Errorf("unable to delete droplet %d: %v", id, err)
 				}
 			}
 			return nil
 		}
-		return matchDroplets(c.Args, ds, fn)
+		return fmt.Errorf("operation aborted")
+
 	}
-	return fmt.Errorf("operation aborted")
 
-	return nil
+	if len(c.Args) < 1 && tagName == "" {
+		return doctl.NewMissingArgsErr(c.NS)
+	} else if len(c.Args) > 0 && tagName != "" {
+		return fmt.Errorf("please specify droplets identifiers or a tag name")
+	} else if tagName != "" {
```

Comment:
Thank you for answer and explanation. Code is working very well. However, this is probably a point where something can be improved.
This part looks OK, but this is not needed for force action. In case of `force`, you don't want to see list and you don't need for list to be generated. In my opinion, you should check is `force` true, and if it is — exit function with `return ds.DeleteByTag(tagName)`. If it's false, execute below code as it is now.
Not something glorious, but it'll save one API call and speed up process (maybe not noticeable) because it doesn't need to query for all Droplets with tag, generate list and go through the `deleteFn` function.

About mocks, I think there is no anything to be improved. There are substituting API actions with desired response so it doesn't return error (like unauthenticated, resource doesn't exist, etc.). There are generated for code in `do` folder using [mockery](https://github.com/vektra/mockery).